### ISSUE TITLE
Include secrets when RUN --secrets is run

### ISSUE THREAD

**Issue by agbell** [earthly/earthly#707]
To run a command with a secret ENV set, I need to specify it twice, once when I run earthly and once when I run the command.  It would be useful if there were a terser syntax.

Now:
```
earthly -P \
    --secret AWS_ACCESS_KEY_ID=foo \
    --secret AWS_SECRET_ACCESS_KEY=bar \
    --secret AWS_SESSION_TOKEN=baz \
```

```
 RUN \
            --secret AWS_ACCESS_KEY_ID=+secrets/AWS_ACCESS_KEY_ID \
            --secret AWS_SECRET_ACCESS_KEY=+secrets/AWS_SECRET_ACCESS_KEY \
            --secret AWS_SESSION_TOKEN=++secrets/AWS_SESSION_TOKEN \
            printenv
```

Potential Simplification:

```
earthly -P \
    --secret AWS_ACCESS_KEY_ID=foo \
    --secret AWS_SECRET_ACCESS_KEY=bar \
    --secret AWS_SESSION_TOKEN=baz \
```

```
 RUN --secrets printenv
```

**Comment by vladaionescu** [comment]
This has the disadvantage that the build is less readable - it's less clear which secrets are actually used.

An alternative, which simplifies the usage is registering [cloud secrets](https://docs.earthly.dev/guides/cloud-secrets). With that, secrets are only referenced in the Earthfile only.

**Comment by kevinkjt2000** [comment]
Cloud secrets may not be a possibility for some projects. Imagine laws/regulations that enforce storage of secrets to particular systems, relying on a cloud for privileged information that is outside of those systems could be illegal.

As for this issue, I don't know how I feel about exposing all secrets to a command when some may want to be more selective about which secrets go to which commands. I concur that it is also less clear which secrets are being used in this way.

For secrets that have exactly the same environment variable name as their counterpart, how about reducing the repetition by changing this
```
 RUN \
            --secret AWS_ACCESS_KEY_ID=+secrets/AWS_ACCESS_KEY_ID \
            --secret AWS_SECRET_ACCESS_KEY=+secrets/AWS_SECRET_ACCESS_KEY \
            --secret AWS_SESSION_TOKEN=+secrets/AWS_SESSION_TOKEN \
            printenv
```
to this
```
 RUN \
            --secret AWS_ACCESS_KEY_ID \
            --secret AWS_SECRET_ACCESS_KEY \
            --secret AWS_SESSION_TOKEN \
            printenv
```
?

**Comment by kevinkjt2000** [comment]
Maybe there needs to be a `SECRET` command? Imagine this scenario:

```
base-aws:
    SECRET AWS_ACCESS_KEY_ID
    SECRET AWS_SECRET_ACCESS_KEY
    SECRET AWS_SESSION_TOKEN
    SECRET HAMMER_TIME=+secrets/CANT_TOUCH_THIS
```

Other targets could then base off of that.

**Comment by brbrown25** [comment]
@kevinkjt2000 I really like those suggestions. I feel like it would be a pretty intuitive and reduce the repetition.

**Comment by vladaionescu** [comment]
> Maybe there needs to be a SECRET command?

What would the behavior be?

Would

```
SECRET ABC
RUN echo $ABC
```

be equivalent to

```
RUN --secret=ABC echo $ABC
```
?

**Comment by kevinkjt2000** [comment]
Exactly what I was picturing, and I'll elaborate further:

```
base-a:
    SECRET X
    RUN printf "X:%s" "$X"

base-b:
    FROM +base-a
    RUN printf "X:%s" "$X"  # X is declared from base-a, so this should work
    RUN printf "Y:%s" "$Y"  #?????? I don't foresee this needing to be supported; secret Y has not been declared yet... I would assume a blank value would be output
    SECRET Y
    RUN printf "X:%s\nY:%s" "$X" "$Y"   #both should work at this location

base-c:
    FROM +base-b
    SECRET X=override
    RUN printf "X:%s\nY:%s" "$X" "$Y"   #I would expect to see X:override printed and Y to be unchanged
```

**Comment by vladaionescu** [comment]
Makes sense - and by `SECRET X=override` assume that override can be a secret path like `+secrets/something/something` and not a literal string (for consistency).

**Comment by vladaionescu** [comment]
Hey @kevinkjt2000 - would you like to work on this issue yourself? I remember that you asked about possible issues to contribute to.

I would start with your [first proposal](https://github.com/earthly/earthly/issues/707#issuecomment-764693208) as the changes will be easier I imagine.

After that, you could add `SECRET` as a separate PR. That one might be a little more difficult. We have this [variables collection](https://github.com/earthly/earthly/blob/main/variables/collection.go) which keeps track of `ENV` and `ARG` declarations. Perhaps `SECRET` can go in there as an additional type, with somewhat similar behavior to `ARG`.

**Comment by andyli** [comment]
Somewhat related, I found the `earthly` command invocation too verbose.

Given a Earthfile with a `RUN` command,
```
 RUN \
            --secret AWS_ACCESS_KEY_ID=+secrets/AWS_ACCESS_KEY_ID \
            --secret AWS_SECRET_ACCESS_KEY=+secrets/AWS_SECRET_ACCESS_KEY \
            --secret AWS_SESSION_TOKEN=++secrets/AWS_SESSION_TOKEN \
            aws s3 ...
```

Can't earthly directly use the host's `AWS_*` env vars, without listing each of them as `earthly --secret AWS_...=...`?
